src/runtime/refiner.ts

```diff
@@ -1621,11 +1621,24 @@ interface OperatorInfo {
 
 const numericTypes: Primitive[] = [Primitive.NUMBER].concat(discreteTypes);
 
+function powBigInt(x: bigint, n: bigint): bigint {
+  switch (n) {
+    case BigInt(0): return BigInt(1);
+    case BigInt(1): return x;
+    default: break;
+  }
+  const k = n/BigInt(2);
+  const d = n%BigInt(2);
+  const x_k = powBigInt(x, k);
+  const x_d = powBigInt(x, d);
+  return x_k*x_k*x_d;
```

Comment:
Forgive my ignorance, but will this work for negative powers?

Reply:
No, but BigInt doesn't support negative powers either. 
BigInt throws a `RangeError: Exponent must be positive` error, I'll update this to do the same.

Reply:
If the only place this is used is to compute those 4 numbers, could we just hardcode them instead?

Reply:
I think it's likely easier to check that this code is right (rather than compute the values by hand when reading over it). I'll make the change and upload it if you reply in the positive / thumbs up this comment.

Reply:
If it's just these 4 numbers, then IMO this is about as readable as it's going to get:

```
const INT_MIN: bigint = -2147483648n; // -2**31
const INT_MAX: bigint = 2147483647n; // 2**31 - 1
const LONG_MIN: bigint = 9223372036854775808n; // -2**63
const LONG_MAX: bigint = 9223372036854775807n; // 2**63 - 1
```